buidl/script.py

```diff
@@ -369,34 +371,92 @@ def convert(cls, raw_redeem_script):
         return cls.parse(stream)
 
     @classmethod
-    def create_p2sh_multisig(cls, quorum_m, pubkey_hex_list, sort_keys=True):
+    def create_p2sh_multisig(
+        cls,
+        quorum_m,
+        pubkey_hexes,
+        sort_keys=True,
+        expected_addr=None,
+        expected_addr_network="mainnet",
+    ):
         """
         Create a p2sh RedeemScript using a configure threshold (quorum_m) of child public keys (in hex).
 
         To use a custom order of pubkeys, feed them in order and set sort_keys=False
+
+        For safety, you may pass in an expected_addr and the method will throw an error if the derived address doesn't match the expected one.
         """
         # safety checks
         if type(quorum_m) is not int:
             raise ValueError(f"quorum_m must be of type int: {quorum_m}")
-        if quorum_m < 1 or quorum_m > len(pubkey_hex_list):
-            raise ValueError(f"Invalid m-of-n: {quorum_m}-of-{len(pubkey_hex_list)}")
+        if quorum_m < 1 or quorum_m > len(pubkey_hexes):
+            raise ValueError(f"Invalid m-of-n: {quorum_m}-of-{len(pubkey_hexes)}")
 
         commands = [number_to_op_code(quorum_m)]
 
         if sort_keys:
-            pubkey_hexes = sorted(pubkey_hex_list)
+            pubkey_hexes = sorted(pubkey_hexes)
         else:
```

Comment:
Is this `else` clause necessary?

Reply:
Good catch, not sure how that got weird. Fixed in https://github.com/buidl-bitcoin/buidl-python/pull/81/commits/a4717e714250a23139e3e1c150ded31ff30f276e